### PR TITLE
fix the string.mli documentation of unsafe-string (not the default anymore)

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

**Comment by gasche** [comment]
(This can be cherry-picked into 4.09.)

**Review by xavierleroy — APPROVED** [review]
Thank you!

**Review by xavierleroy — COMMENTED** [review]
Actually, the next paragraph may need rewording too:

All new code should avoid this feature and be compiled with the -safe-string command-line option to enforce the separation between the types string and bytes.

**Comment by gasche** [comment]
Yes, I sensed that something would have to change in this PR :-) 
I replaced the second paragraph with the following:

> The difference between [bytes] and [string] was introduced in OCaml
> 4.02, and the "unsafe-string" compatibility mode was the default
> until OCaml 4.05. Starting with 4.06, the compatibility mode is
> opt-in; we are planning to remove the option in the future.

File: stdlib/string.mli [hunk 1]
```diff
@@ -32,21 +32,19 @@
...
- *)
+  Note: OCaml strings used to be modifiable in place, for instance via
+  the {!String.set} and {!String.blit} functions described below. This
+  usage is only possible when the compiler is put in "unsafe-string"
+  mode by giving the [-unsafe-string] command-line option. This
+  compatibility mode makes the types [string] and [bytes] (see module
+  {!Bytes}) interchangeable so that functions expecting byte sequences
+  can also accept strings as arguments and modify them.
+
+  The difference between [bytes] and [string] was introduced in OCaml
```
Comment (xavierleroy) [inline stdlib/string.mli]: Looks good.  I wonder what word is better: "The difference between" or "The distinction between"...  Any native English speaker around?

File: stdlib/string.mli [hunk 2]
```diff
@@ -32,21 +32,19 @@
...
+  4.02, and the "unsafe-string" compatibility mode was the default
+  until OCaml 4.05. excluded). Starting with 4.06, the compatibility
```
Comment (damiendoligez) [inline stdlib/string.mli]: excluded? You mean "included", right?

**Review by damiendoligez — COMMENTED** [review]
Let's be wild and cherry-pick to 4.08 too.

File: stdlib/string.mli (hunk 1 above)
Comment (jberdine) [reply stdlib/string.mli]: FWIW I'd use 'distinction' since there is for now still an equality underneath.

File: stdlib/string.mli (hunk 2 above)
Comment (dra27) [reply stdlib/string.mli]:
```suggestion
  until OCaml 4.05 inclusive). Starting with 4.06, the compatibility
```

File: stdlib/string.mli [hunk 3]
```diff
@@ -32,21 +32,19 @@
...
+  mode is opt-in; we are planning to remove the option in the future.
```
Comment (dra27) [inline stdlib/string.mli]: Could perhaps be stronger and "we intend to remove the option."

File: stdlib/string.mli (hunk 2 above)
Comment (gasche) [reply stdlib/string.mli]: This was an edit error on my part. I started with "until 4.06 (excluded)", and that felt too formal so I am now using the long form "until 4.05. In 4.06, ...", which is more explicit.

File: stdlib/string.mli (hunk 3 above)
Comment (gasche) [reply stdlib/string.mli]: Done, thanks!

File: stdlib/string.mli (hunk 2 above)
Comment (gasche) [reply stdlib/string.mli]: (Fixed.)

File: stdlib/string.mli (hunk 1 above)
Comment (gasche) [reply stdlib/string.mli]: I went for 'distinction', thanks!

**Review by xavierleroy — APPROVED** [review]
Looks very good now, thanks all for the proofreading and @gasche for the editing.  Now: merging!

**Comment by xavierleroy** [comment]
Cheery-picked in 4.09 (cb73ea1) and in 4.08 (c6c9c72).

**Pull request merged**